pkg/scheduler/scheduler_test.go

```diff
@@ -1651,6 +1802,428 @@ func TestSchedule(t *testing.T) {
 				"eng-gamma/gamma4":   *utiltesting.MakeAdmission("eng-gamma").Assignment(corev1.ResourceCPU, "on-demand", "20").Obj(),
 			},
 		},
+		"multiple preemptions without borrowing": {
+			// While requiring the same shared FlavorResource (Default, cpu),
+			// multiple workloads are able to issue preemptions on workloads within
+			// their own CQs in a single scheduling cycle.
+			multiplePreemptions: MultiplePremptions,
```

Comment:
can't this be left for both?

Reply:
Nope, the previous logic would trigger the skip preemption logic, as it counted overlapping FlavorResource.

Reply:
we added a test case to cover this for the previous logic in https://github.com/kubernetes-sigs/kueue/pull/2646

Reply:
oh, true, true, this is what we are trying to fix lol